tests/conftest.py

```diff
@@ -43,14 +44,34 @@ def get_database_url():
 
 
 @pytest.fixture(scope="session")
-def db_engine():
+def db_engine(tmp_path_factory):
     engine = sqlalchemy.create_engine(TEST_SETTINGS["database_url"])
 
-    # Delete all database tables and re-initialize the database schema based on
-    # the current models. Doing this at the beginning of each test run ensures
-    # that any schema changes made to the models since the last test run will
-    # be applied to the test DB schema before running the tests again.
-    db.init(engine, drop=True, stamp=False)
+    # Use a filelock to only init the DB once even though we have multiple
+    # parallel pytest-xdist workers. See:
+    # https://pytest-xdist.readthedocs.io/en/stable/how-to.html?highlight=filelock#making-session-scoped-fixtures-execute-only-once
+
+    # The temporary directory shared by all pytest-xdist workers.
+    shared_tmpdir = tmp_path_factory.getbasetemp().parent
+
+    # The existence of this file records that a worker has initialized the DB.
+    done_file = shared_tmpdir / "db_initialized"
```

Comment:
If we are cookie cutting this maybe it could be clearer declaring here both files:

```
# The presence of this file prevents this fixutre to be executed by different xdist workers simultaneously  
lock_file = str(done_file) + ".lock"
```